### ISSUE TITLE
Bing api subscription key problem

### ISSUE THREAD

**Issue by chriswessells** [lanmaster53/recon-ng#132]
I could not figure out how to create a subscription key for the cognitive services search. Links to the azure marketplace option  didn't work. 

I read through the bing search documentation and created a subscription key for a Bing Resource [1]. The subscription key didn't work.

I was able to get everything to work when I updated the url for the bing search api in mixins serach_bing_api. 

url = 'https://api.bing.microsoft.com/v7.0/search'

I'm happy to submit a pull request with an update to the url but I'm not sure if I'm just doing something wrong? I wanted to share what I found :) 

[1] https://docs.microsoft.com/en-us/azure/cognitive-services/bing-web-search/overview

**Comment by rsalvalagio** [comment]
Hi Chris, I'm facing the same issue, but I can't find the mixins directory containing search_bing_api.py. Even if I use find / -name, it didn't appear in the search results. Can you please write the full path you used?

**Comment by robbie888** [comment]
Hi,
I had the same problem, but I needed to change the URL in the mixins search script, otherwise I kept getting a key error.

When I got my API key (which was hard enough to find in the Azure page!) it showed me an 'endpoint' URL under the API key which is what I assume is the newest url.

When I updated that in the recon search.py script it stopped giving me the key error.

The file I had to edit was: `/usr/share/recon-ng/recon/mixins/search.py`
I replaced `https://api.bing.microsoft.com/v7.0/search` with  `https://api.bing.microsoft.com`

For those struggling to find how to get a key, in the Azure webportal go to marketplace and search for 'bing search', there you will see Bing Search v7, sign up to that (free version) and you'll get a key pretty quickly. 

Hope that helps someone else out!
